activerecord/test/cases/relation/or_test.rb

```diff
@@ -174,6 +174,17 @@ def test_structurally_incompatible_values
         Post.from("posts").or(Post.from("posts"))
       end
     end
+
+    def test_or_with_random_ordered_multiple_values
+      assert_nothing_raised do
+        Post.includes(:author, :comments).or(Post.includes(:comments, :author))
+        Post.eager_load(:author, :comments).or(Post.eager_load(:comments, :author))
+        Post.preload(:author, :comments).or(Post.preload(:comments, :author))
+        Post.joins(:author, :comments).or(Post.joins(:comments, :author))
+        Post.left_outer_joins(:author, :comments).or(Post.left_outer_joins(:comments, :author))
+        Post.select(:body, :title).or(Post.select(:title, :body))
```

Comment:
This assertion is asserting the whole block rather than each of these calls. Asserting that nothing is raised doesn't tell us much about what our expectations are. These assertions should instead assert the return values for each call. If something is raised then the test will fail.